apl-utils/src/test/java/com/apollocurrency/aplwallet/apl/util/task/BackgroundTaskDispatcherTest.java

```diff
@@ -0,0 +1,174 @@
+package com.apollocurrency.aplwallet.apl.util.task;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.concurrent.RejectedExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@Slf4j
+@ExtendWith(MockitoExtension.class)
+@Execution(ExecutionMode.CONCURRENT)
+class BackgroundTaskDispatcherTest {
+
+    private TaskDispatcher taskDispatcher;
+    private Runnable runnable;
+    private Task task;
+
+    @BeforeEach
+    void setUp() {
+        runnable = mock(Runnable.class);
+        task = Task.builder()
+                .name("TestTask")
+                .delay(10)
+                .initialDelay(0)
+                .task(runnable)
+                .build();
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    void scheduleAtFixedRate() {
+        taskDispatcher = TaskDispatcherFactory.newScheduledDispatcher("TestThreadInfo");
+        task.setTask(runnable);
+
+        taskDispatcher.schedule(task);
+
+        taskDispatcher.dispatch();
+        log.info("Thread dispatch");
+
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException ignored) {}
```

Comment:
When sleep will be interrupted, your test will fail due to suppressed exception

Reply:
added code to set interrupted flag